core/collection/cache.go

```diff
@@ -144,7 +146,9 @@ func (c *Cache) Take(key string, fetch func() (any, error)) (any, error) {
 		return v, nil
 	})
 	if err != nil {
-		return nil, err
+		return val, err
+	} else if val, typeConvertFlag = _val.(T); !typeConvertFlag {
+		return val, ErrTypeConvert
```

Comment:
No need `ErrTypeConvert`, users should ensure that there should not be more than one fetch function for the same key, so let it panic.